bouncer/src/repo/manipulator/modelconvertor/export/repo_model_export_asset.cpp

```diff
@@ -233,19 +233,22 @@ bool AssetModelExport::generateTreeRepresentation()
 			superMeshesTree.addToTree("model", scene->getProjectName());
 			superMeshesTree.addArrayObjects("supermeshes", meshesTrees);
 
-			jsonTrees["/revision/" + scene->getRevisionID().toString() + "/supermeshes.json"] = superMeshesTree;
+			jsonTrees["/revision/" + scene->getRevisionID().toString() + "/supermeshes_unity.json"] = superMeshesTree;
```

Comment:
It shouldn't: this file (`supermeshes_unity.json`) is the container-wide cache file (not the regular mappings file). 

In "normal" legacy mode (i.e existing revisions), they will request `supermeshes.json`.

`supermeshes_unity.json` should only be read if the user is operating in forced-legacy mode. In this case yes it will indeed not exist for existing revisions, but .io will re-create it on demand, as it did for models before we added the cache.

The `generateSuperMeshMappings` function in .io has not changed, since all post-RepoBundles containers should be imported with the cache files already existing, and the existing unityAssets.json should be backwards-compatible.

(The idea is that in a couple of releases when it goes away, we are left with just one API that works with new and existing cache files transparently, without having to know which type of bundle is underneath.)

Reply:
ok I don't think I fully understand this yet... 😆 

so supermeshes_unity.json store different information to the old supermeshes.json and the latter is no longer used with repo bundles, and we no longer generate the old `supermeshes.json`?

So this becomes an issue if we have to rollback, with new models it will try to load supermeshes.json which doesn't exist and 3drepo.io will have to generate the cache on the fly?

(the 3drepo.io generate cache is never ran live - it was there so I can run it through a script 🙈 the problem was that this cache was taking upwards of 6mins to run so it becomes a problem)

I suppose there's no easy way to write both for now and remove the old one in the future?

Reply:
> (the 3drepo.io generate cache is never ran live - it was kept there so I can run it through a script 🙈 the problem was that this cache was taking upwards of 6mins to run so it becomes a problem)

OK I didn't realise that!

> so supermeshes_unity.json store different information to the old supermeshes.json and the latter is no longer used with repo bundles, and we no longer generate the old supermeshes.json?

No I haven't explained it well 😄 

The supermeshes.json (cache) continues to be generated and used by RepoBundles, just like it was for AssetBundles. Whereas before it was built by `AssetModelExport`, its now built by `RepoBundleExport`. There is only one file and one API for new and existing imports, and the viewer handles the existing versions transparently.

This is possible because the submeshid layouts are identical between AssetBundles and RepoBundles, so the same Mappings files apply to both. The only difference is the old ones don't have additional material data the `RepoBundles` expect.

Therefore, the idea going forward is that the viewer has one pathway for handling the mappings, and only the Bundles themselves have dedicated import functions.

This is for maintainability. The function to handle the mappings is quite complex, and the only difference between old and new is the additional bit of logic to handle materials. Further, when we get to Milestone 2 proper, its likely the logic in this area will undergo significant refactoring, so we ideally don't want to indefinitely look after two paths that are 95% identical.

In fact, the original implementation had just one set of Jsons in unityAssets.

However, I split them out in unityAssets, and added the supermeshes_unity.json above, so that bouncer would continue to generate complete mappings independently of RepoBundles, in case there turned out to be an issue.

I can't think of what that might be, but then if I could be 100% sure there are no problems with RepoBundles we wouldn't need to run them side by side for a time!

In brief then, this file is a backup for a backup: if there's a problem with RepoBundles we should in theory be able to fall back to AssetBundles + new mappings. However, if there is a problem with RepoBundles AND the new mappings, we can fallback to AssetBundles + supermeshes_unity.json.

Reply:
ok, fair enough 👍